Windows compilation error: cannot allocate an array of constant size 0

Hello,

I'm reporting a compilation error we get on windows using vs2019.

Full log attached.

```
[21/24] Building C object tests\CMakeFiles\check_rsi_block_access.dir\check_rsi_block_access.c.obj
FAILED: tests/CMakeFiles/check_rsi_block_access.dir/check_rsi_block_access.c.obj 
C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\HostX64\x64\cl.exe  /nologo  -I%SRC_DIR%\src\..\include -I%SRC_DIR%\build\src\..\include /DWIN32 /D_WINDOWS /O2 /Ob2 /DNDEBUG -MD /showIncludes /Fotests\CMakeFiles\check_rsi_block_access.dir\check_rsi_block_access.c.obj /Fdtests\CMakeFiles\check_rsi_block_access.dir\ /FS -c %SRC_DIR%\tests\check_rsi_block_access.c
%SRC_DIR%\tests\check_rsi_block_access.c(46): warning C4477: 'fprintf' : format string '%lu' requires an argument of type 'unsigned long', but variadic argument 1 has type 'size_t'
%SRC_DIR%\tests\check_rsi_block_access.c(46): note: consider using '%zu' in the format string
%SRC_DIR%\tests\check_rsi_block_access.c(68): warning C4477: 'fprintf' : format string '%lu' requires an argument of type 'unsigned long', but variadic argument 1 has type 'size_t'
%SRC_DIR%\tests\check_rsi_block_access.c(68): note: consider using '%zu' in the format string
%SRC_DIR%\tests\check_rsi_block_access.c(94): warning C4477: 'fprintf' : format string '%lu' requires an argument of type 'unsigned long', but variadic argument 1 has type 'size_t'
%SRC_DIR%\tests\check_rsi_block_access.c(94): note: consider using '%zu' in the format string
%SRC_DIR%\tests\check_rsi_block_access.c(318): error C2057: expected constant expression
%SRC_DIR%\tests\check_rsi_block_access.c(318): error C2466: cannot allocate an array of constant size 0
%SRC_DIR%\tests\check_rsi_block_access.c(318): error C2133: 'slice_offsets': unknown size
%SRC_DIR%\tests\check_rsi_block_access.c(319): error C2057: expected constant expression
%SRC_DIR%\tests\check_rsi_block_access.c(319): error C2466: cannot allocate an array of constant size 0
%SRC_DIR%\tests\check_rsi_block_access.c(319): error C2133: 'slice_sizes': unknown size
[22/24] Linking C executable tests\check_szcomp.exe
ninja: build stopped: subcommand failed.
```

[36.txt](https://github.com/MathisRosenhauer/libaec/files/12777438/36.txt)

--- a/tests/check_rsi_block_access.c
+++ b/tests/check_rsi_block_access.c
@@ -43,7 +43,7 @@ static void data_generator_zero(struct aec_context *ctx)
 {
     size_t nbytes = ctx->bytes_per_sample;
     if (ctx->obuf_len % nbytes) {
-        fprintf(stderr, "Invalid buffer_size: %lu\n", ctx->obuf_len);
+        fprintf(stderr, "Invalid buffer_size: %zu\n", ctx->obuf_len);
         exit(1);
     }
 
@@ -65,7 +65,7 @@ static void data_generator_random(struct aec_context *ctx)
 {
     size_t nbytes = ctx->bytes_per_sample;
     if (ctx->obuf_len % nbytes) {
-        fprintf(stderr, "Invalid buffer_size: %lu\n", ctx->obuf_len);
+        fprintf(stderr, "Invalid buffer_size: %zu\n", ctx->obuf_len);
         exit(1);
     }
 
@@ -91,7 +91,7 @@ static void data_generator_incr(struct aec_context *ctx)
 {
     size_t nbytes = ctx->bytes_per_sample;
     if (ctx->obuf_len % nbytes) {
-        fprintf(stderr, "Invalid buffer_size: %lu\n", ctx->obuf_len);
+        fprintf(stderr, "Invalid buffer_size: %zu\n", ctx->obuf_len);
         exit(1);
     }
 
@@ -315,8 +315,8 @@ int test_read(struct aec_context *ctx)
     size_t num_slices = ctx->obuf_len / slice_size;
     size_t remainder = ctx->obuf_len % slice_size;
 
-    size_t slice_offsets[num_slices + 1];
-    size_t slice_sizes[num_slices + 1];
+    size_t *slice_offsets = malloc((num_slices + 1) * sizeof(slice_offsets[0])); 
+    size_t *slice_sizes = malloc((num_slices + 1) * sizeof(slice_sizes[0]));
 
     for (size_t i = 0; i < num_slices; ++i) {
         slice_offsets[i] = slice_size * i;
@@ -385,6 +385,8 @@ int test_read(struct aec_context *ctx)
     aec_decode_end(&strm_read);
 
     free(offsets);
+    free(slice_offsets);
+    free(slice_sizes);
     return status;
 }
 

